packages/pnpm-sync-lib/src/pnpmSyncCopy.ts

```diff
@@ -93,17 +99,34 @@ export async function pnpmSyncCopyAsync(options: IPnpmSyncCopyOptions): Promise<
 
   const pnpmSyncJsonFolder = path.dirname(pnpmSyncJsonPath);
 
-  //read the .pnpm-sync.json
-  const pnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+  // read the .pnpm-sync.json
+  const pnpmSyncJson: IPnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+
+  // verify if the version is outdated
+  const pnpmSyncJsonVersion: string = getPnpmSyncJsonVersion();
+
+  if (pnpmSyncJsonVersion !== pnpmSyncJson.version) {
+    const errorMessage = `The .pnpm-sync.json file in ${pnpmSyncJsonFolder} is outdated; regenerate it and try again.`;
```

Comment:
```suggestion
    const errorMessage = `The .pnpm-sync.json file in ${pnpmSyncJsonFolder} has an incompatible version; regenerate it and try again.`;
```

The version might be **_too new_** or too old.

This message is vague about what the user is supposed to do.  Rush's implementation should customize this message to suggest invoking `rush update` or `rush install`.